Failed build with librabbitmq 0.5

1.6.0beta3 build fine with librabbitmq 0.5.x (available in EPEL for RHEL and CentOS, can't be upgrade because of ABI break)

Buildtime:

    /builddir/build/BUILD/more-php56-php-pecl-amqp-1.6.0/NTS/amqp_connection.c:146: warning: implicit declaration of function 'amqp_get_frame_max'
    /builddir/build/BUILD/more-php56-php-pecl-amqp-1.6.0/NTS/amqp_connection.c:154: warning: implicit declaration of function 'amqp_get_heartbeat'

Runtime:

     PHP Warning:  PHP Startup: Unable to load dynamic library 'NTS/modules/amqp.so' - NTS/modules/amqp.so: undefined symbol: amqp_get_heartbeat in Unknown on line 0

It will be nice to have this feature conditional.
If not possible, to check (in config.m4) with a message explaining minimal version required.

--- a/README.md
+++ b/README.md
@@ -6,7 +6,8 @@ Object-oriented PHP bindings for the AMQP C library (https://github.com/alanxz/r
 ### Requirements:
 
  - [RabbitMQ C library](https://github.com/alanxz/rabbitmq-c), commonly known as librabbitmq
-   (since php-amqp>=1.6.0  librabbitmq >= 0.6.0 required, see [release note](http://pecl.php.net/package/amqp/1.6.0)).
+   (since php-amqp>=1.6.0  librabbitmq >= 0.5.2 required, >= 0.6.0 recommended,
+   see [release note](http://pecl.php.net/package/amqp/1.6.0)).
 
 ### Installation
  

--- a/amqp.c
+++ b/amqp.c
@@ -493,6 +493,10 @@ zend_function_entry amqp_connection_class_functions[] = {
 	PHP_ME(amqp_connection_class, getUsedChannels, arginfo_amqp_connection_class_getUsedChannels,	ZEND_ACC_PUBLIC)
 	PHP_ME(amqp_connection_class, getMaxChannels,  arginfo_amqp_connection_class_getMaxChannels,	ZEND_ACC_PUBLIC)
 	PHP_ME(amqp_connection_class, isPersistent, 	arginfo_amqp_connection_class_isPersistent,		ZEND_ACC_PUBLIC)
+#if AMQP_VERSION_MAJOR * 100 + AMQP_VERSION_MINOR * 10 + AMQP_VERSION_PATCH > 52
+	PHP_ME(amqp_connection_class, getHeartbeatInterval,  arginfo_amqp_connection_class_getHeartbeatInterval,	ZEND_ACC_PUBLIC)
+	PHP_ME(amqp_connection_class, getMaxFrameSize,  arginfo_amqp_connection_class_getMaxFrameSize,	ZEND_ACC_PUBLIC)
+#endif
 
 	{NULL, NULL, NULL}	/* Must be the last line in amqp_functions[] */
 };

--- a/amqp_connection.c
+++ b/amqp_connection.c
@@ -141,6 +141,7 @@ HashTable *amqp_connection_object_get_debug_info(zval *object, int *is_temp TSRM
 	}
 	zend_hash_add(debug_info, "max_channel_id", sizeof("max_channel_id"), &value, sizeof(zval *), NULL);
 
+#if AMQP_VERSION_MAJOR * 100 + AMQP_VERSION_MINOR * 10 + AMQP_VERSION_PATCH > 52
 	MAKE_STD_ZVAL(value);
 	if (connection->connection_resource) {
 		ZVAL_LONG(value, amqp_get_frame_max(connection->connection_resource->connection_state));
@@ -156,6 +157,7 @@ HashTable *amqp_connection_object_get_debug_info(zval *object, int *is_temp TSRM
 		ZVAL_NULL(value);
 	}
 	zend_hash_add(debug_info, "heartbeat_interval", sizeof("heartbeat_interval"), &value, sizeof(zval *), NULL);
+#endif
 
 	/* Start adding values */
 	return debug_info;
@@ -1361,6 +1363,7 @@ PHP_METHOD(amqp_connection_class, getMaxChannels)
 }
 /* }}} */
 
+#if AMQP_VERSION_MAJOR * 100 + AMQP_VERSION_MINOR * 10 + AMQP_VERSION_PATCH > 52
 /* {{{ proto amqp::getMaxFrameSize()
 Get max supported frame size per connection in bytes */
 PHP_METHOD(amqp_connection_class, getMaxFrameSize)
@@ -1408,6 +1411,7 @@ PHP_METHOD(amqp_connection_class, getHeartbeatInterval)
 	RETURN_LONG(amqp_get_heartbeat(connection->connection_resource->connection_state));
 }
 /* }}} */
+#endif
 
 /* {{{ proto amqp::isPersistent()
 check whether amqp connection is persistent */

--- a/amqp_connection.h
+++ b/amqp_connection.h
@@ -68,9 +68,11 @@ PHP_METHOD(amqp_connection_class, getWriteTimeout);
 PHP_METHOD(amqp_connection_class, setWriteTimeout);
 
 PHP_METHOD(amqp_connection_class, getUsedChannels);
-PHP_METHOD(amqp_connection_class, getMaxFrameSize);
 PHP_METHOD(amqp_connection_class, getMaxChannels);
+#if AMQP_VERSION_MAJOR * 100 + AMQP_VERSION_MINOR * 10 + AMQP_VERSION_PATCH > 52
 PHP_METHOD(amqp_connection_class, getHeartbeatInterval);
+PHP_METHOD(amqp_connection_class, getMaxFrameSize);
+#endif
 PHP_METHOD(amqp_connection_class, isPersistent);
 
 /*

